.github/workflows/test.yml

```diff
@@ -0,0 +1,36 @@
+name: Test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Setup Node
+      uses: actions/setup-node@v3.1.1
+      with:
+        node-version: 16.13
+        cache: npm
+        
+    - name: Rust toolchain
+      uses: actions-rs/toolchain@v1
```

Comment:
what's the toolchain used for `v1` version here? which version is more preferred?

Reply:
i think this is the github rust action version, v1 is the only one I can find.
Actually without this action the job could still run, I added this because it claims it has cache support